irods/library/irods_move

```diff
@@ -1,142 +1,157 @@
+
+author:
+    - Fenn Garnett (@Fennersteel)
+'''
+
+EXAMPLES = '''
+# Move iRODS data object/collection
+- name: move data object/collection
+    irods_move:
+        source: /testing/home/rods/test_src
+        destination: /testing/home/rods/test_dst
+        host: "{{ groups['irods_catalog'][0] }}"
+        port: 1247
+        admin_user: rods
+        admin_password: rods
+        zone: testing
+'''
+
+RETURN = '''
+exc:
+    description:
+        - type of last iRODS exception thrown
+        - empty string when none is thrown
+    type: str
+    returned: always
+exc_msg:
+    description:
+        - message of the last iRODS exception thrown
+        - empty string when none is thrown
+    type: str
+    returned: always
+'''
+
+
+class IRODSMoveModule:
+    """
+    Module class
+    """
+
+    def __init__(self):
+        """
+        Initialize module
+        """
+
+        self.module_args = dict(
+            source=dict(type="str", required=True),
+            destination=dict(type="str", required=True),
+
+            host=dict(type="str", required=True),
+            port=dict(type="int", required=True),
+            admin_user=dict(type="str", no_log=True, required=True),
+            admin_password=dict(type="str", no_log=True, required=True),
+            zone=dict(type="str", required=True),
+        )
+
+        self.result = dict(
+            changed=False,
+            exc='',
+            exc_msg=''
+        )
+
+        self.module = AnsibleModule(
+            argument_spec=self.module_args,
+            supports_check_mode=True
+        )
+
+        self.ssl_context = ssl.create_default_context(
+            purpose=ssl.Purpose.SERVER_AUTH, cafile=None, capath=None,
+            cadata=None)
+
+        self.ssl_settings = {"ssl_context": self.ssl_context}
+
+        self.session = iRODSSession(
+            source=self.module.params["source"],
+            destination=self.module.params["destination"],
+            host=self.module.params["host"],
+            port=self.module.params["port"],
+            user=self.module.params["admin_user"],
+            password=self.module.params["admin_password"],
+            zone=self.module.params["zone"],
+            **self.ssl_settings)
+
+    def run_module(self):
+        if self.module.check_mode:
+            self.module.exit_json(**self.result)
+
+        try:
+            self.rename()
+            self.result["changed"] = True
+            self.module.exit_json(**self.result)
+        except Exception as exc:
+            self.result["exc"] = type(exc).__name__
+            self.result["exc_msg"] = str(exc)
+            self._fail("Unable to move irods object/collection {}".format(
+                self.module.params["source"]), exc)
+
+    def rename(self):
+        # rename collection
+        path = self.module.params["source"]
+        new_path = self.module.params["destination"]
+        self.session.collections.move(path, new_path)
```

Comment:
This function needs to verify that `path` exists and `new_path` does not before calling `session.collections.move`. If `path` does not exist, and  `new_path` does, then it should assume the move has already happened and the `changed` result left as `False`.